core/currentLimiter.go

```diff
@@ -0,0 +1,121 @@
+package core
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/evcc-io/evcc/util"
+)
+
+// currentLimiter abstracts the LM currentLimiter implementation
+// TBD: required?
+// type currentLimiter interface {
+// 	Enable(enable bool)                                             // enable/disable limitting
+// 	SetTotalLimit(current float64)                                  // set the total limit of this limiter. 0 = unlimited
+// 	SetMargin(current float64)                                      // sets the limit by providing the margin to current consumption
+// 	EvaluateConsumption() float64                                   // evaluates the actual current consumption of
+// 	RequestCurrent(lp *LoadPoint, currentRequested float64) float64 // request current, return limit <= currentRequested
+// }
+
+type CurrentLimiter struct {
+	title      string // meaningful name
+	log        *util.Logger
+	uiChan     chan<- util.Param
+	enabled    bool         // apply limitting
+	maxCurrent float64      // the max allowed current of this limiter
+	loadpoints []*LoadPoint // all loadpoints under management
```

Comment:
Besser wäre hier ein klares Interface. Der Limiter soll nicht in Interna eingreifen.

Reply:
Was wäre denn das klare Interface? Der limiter benutzt die LP um den aktuellen Verbrauch zu ermitteln (wenn kein Meter dafür vorhanden ist).
`type Consumer interface {GetCurrents()[]float64}`?